opentelemetry/src/metrics/mod.rs

```diff
@@ -22,12 +22,12 @@ pub use instruments::{
 pub use meter::{Meter, MeterProvider};
 
 /// A specialized `Result` type for metric operations.
-pub type Result<T> = result::Result<T, MetricsError>;
+pub type MetricResult<T> = result::Result<T, MetricError>;
 
 /// Errors returned by the metrics API.
 #[derive(Error, Debug)]
 #[non_exhaustive]
-pub enum MetricsError {
+pub enum MetricError {
```

Comment:
With these changes, `MetricReader`, `MetricError`, and `MetricResult` are singular, while PushMetricsExporter is plural. Should we PushMetricsExporter as separate PR ?

Reply:
Yeah the end goal will be have that one as singular too, I was going to open another PR for those changes, keeping this one scoped to the result and error types